.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,23 @@
+name: Style Check
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  pre-commit:
+    name: Pre-commit checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: set PY
+        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
```

Comment:
I'm guessing this is to speed up the CI?

Reply:
Yes, it just caches the stuff that pre-commit downloads. I think it creates a venv for each hook. The key makes sure that in case the Python version changes or `.pre-commit-config.yaml`, we don't use the cache, but regenerate. This helps in CI to avoid clashes between different branches that change any of the two. If they all hit the same cache entry, they would keep invalidating the cache.